ugvc/pipelines/vcfeval_flavors.py

```diff
@@ -0,0 +1,113 @@
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+# DESCRIPTION
+#    Compare vcf file to ground-truth in alternative manner
+# CHANGELOG in reverse chronological order
+
+from __future__ import annotations
+import argparse
+import subprocess
+import json
+import pysam
+
+from simppl.simple_pipeline import SimplePipeline
+from ugvc.utils.stats_utils import get_precision, get_recall, get_f1
+
+
+def get_parser(argv: list[str]):
+    ap = argparse.ArgumentParser(prog='vcfeval_flavors.py', description=run.__doc__)
+    ap.add_argument('-b', '--baseline',
+                        help="VCF file containing baseline variants", type=str, required=True)
+    ap.add_argument('-c', '--calls',
+                        help="VCF file containing called variants--output_prefix", type=str, required=True)
+    ap.add_argument('-e', '--evaluation_regions',
+                        help='if set, evaluate within regions contained in the supplied BED file, '
+                             'allowing transborder matches. To be used for truth-set', type=str)
+    ap.add_argument('-o', '--output',
+                        help='directory for output', type=str, required=True)
+    ap.add_argument('-t', '--template',
+                        help='SDF of the reference genome the variants are called against')
+    ap.add_argument('-p', '--allele_and_genotype_error_penalty',
+                        type=int,
+                        choices=[2, 1, 0, -1],
+                        default=0,
+                        help="-p 2: usual vcfeval, penalizes twice each wrong-allele/genotype errror\n" \
+                             "-p 1: penalize wrong-allele/genotype only once (remove 0.5 of such fps and fns)\n" \
+                             "-p 0: don't penalize wrong-allele/genotype (remove completely such fps and fns)\n" \
+                             "-p -1: don't penalize, and also reward a tp for calling wrong-allele/genotype\n")
+    return ap
+    return args
+
+def count_vcf_lines(vcf_file):
+    v_iter = pysam.VariantFile(vcf_file)
+    count = 0
+    for variant in v_iter:
+        count += 1
+    return count
+
+def run(argv: list[str]):
+    """Evaluate VCF against baseline, giving alternative penalty to wrong-alleles and genotype errors"""
+    parser = get_parser(argv)
+    SimplePipeline.add_parse_args(parser)
+    args = parser.parse_args(argv[1:])
+    sp = SimplePipeline(args.fc, args.lc, args.d)
+    calls = args.calls
+    baseline = args.baseline
+    eval_reg = args.evaluation_regions
+    sdf = args.template
+    out_dir = args.output
+    penalty = args.allele_and_genotype_error_penalty
+    result = []
+    sp.print_and_run(f'rm -rf {out_dir}')
+    sp.print_and_run(f'rtg vcfeval -b {baseline} -c {calls} -e {eval_reg} -t {sdf} -o {out_dir}')
+    sp.print_and_run(f'bedtools intersect -a {calls} -b {eval_reg} -header > {out_dir}/input_in_hcr.vcf')
```

Comment:
I would use bcftools view with the -R option. Probably the same, but it is best to stay in the bcftools framework. It handles weird edge cases better.

Reply:
ok, will try it.